pkg/kube/configmap.go

```diff
@@ -0,0 +1,82 @@
+package kube
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/IBM/argocd-vault-plugin/pkg/vault"
+	corev1 "k8s.io/api/core/v1"
+	k8yaml "sigs.k8s.io/yaml"
+)
+
+// ConfigMapTemplate is the template for Kubernetes ConfigMaps
+type ConfigMapTemplate struct {
+	Resource
+}
+
+// NewConfigMapTemplate returns a *ConfigMapTemplate given the template's data, and a VaultType
+func NewConfigMapTemplate(template map[string]interface{}, vault vault.VaultType) (*ConfigMapTemplate, error) {
+	path := os.Getenv("VAULT_PATH_PREFIX")
+	data, err := vault.GetSecrets(path)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ConfigMapTemplate{
+		Resource{
+			templateData: template,
+			vaultData:    data,
+		},
+	}, nil
+}
+
+// Replace will replace the <placeholders> in the template's data with values from Vault.
+// It will return an aggregrate of any errors encountered during the replacements
+func (d *ConfigMapTemplate) Replace() error {
+
+	// Assuming all other fields of a configMap (besides `data`) will have string values
+	replaceInner(&d.Resource, &d.templateData, configMapReplacement)
+	if len(d.replacementErrors) != 0 {
+
+		// TODO format multiple errors nicely
+		return fmt.Errorf("Replace: could not replace all placeholders in ConfigMapTemplate: %s", d.replacementErrors)
+	}
+	return nil
+}
+
+func configMapReplacement(key, value string, vaultData map[string]interface{}) (interface{}, []error) {
+	res, err := genericReplacement(key, value, vaultData)
+	if err != nil {
+		return nil, err
+	}
+	// configMap data values must be strings
+	switch res.(type) {
```

Comment:
I saw this in the secret method, can this be abstracted into a util method?

Reply:
Yeah I think so - in secret it turns the `interface{}` into a byte string, and all strings (which is what this does) can be made into byte strings. Let me do that